lsassy/output/pretty_output.py

```diff
@@ -6,34 +6,125 @@ class Output(IOutput):
     """
     Return output in pretty colorful format
     """
-    def get_output(self):
+
+    def get_output(self) -> str:
+        """
+        Retrieve and format the credentials information in a readable and aligned manner.
+
+        Returns:
+            str: A string representing the formatted output containing the credentials information.
+                If no credentials are found, it returns a message indicating so.
+        """
+        # Step 1: Initialize an empty list to store the final output.
         output = []
-        if len(self._credentials) == 0:
-            return "No credentials found"
-        else:
-            max_size = max(len(c["domain"]) + len(c["username"]) for c in self._credentials)
-            credentials = []
-            for cred in self._credentials:
-                cred_type = "PWD"
-                if cred["password"] is None:
-                    if cred["ticket"] is not None:
-                        cred["password"] = "Domain: {} - End time: {} ({})".format(cred["ticket"]["domain"], cred["ticket"]["endtime"].strftime("%Y-%m-%d %H:%M"), cred['ticket']['file'])
-                        cred_type = "TGT"
-                    elif cred["masterkey"] is not None:
-                        cred["password"] = "{}".format(cred["masterkey"])
-                        cred_type = "DPAPI Masterkey"
-                    else:
-                        cred["password"] = ':'.join(h for h in [cred["lmhash"], cred["nthash"]] if h is not None)
-                        cred_type = "NT"
-                if [cred["domain"], cred["username"], cred["password"]] not in credentials:
-                    credentials.append([cred["domain"], cred["username"], cred["password"]])
-                    output.append(
-                        "{}{}{}{}{}{}".format(
-                            ('{}\\'.format(cred["domain"]) if cred["domain"] is not None and cred["domain"] != "" else " "),
-                            cred["username"],
-                            " " * (max_size - len(cred["domain"]) - len(cred["username"]) + 2),
-                            lsassy_logger.lsassy_highlight("[{}] ".format(cred_type)),
-                            lsassy_logger.lsassy_highlight(cred["password"]),
-                            " | {}".format(lsassy_logger.lsassy_highlight("[{}] {}".format("SHA1", cred["sha1"]))) if cred["sha1"] else "")
-                    )
+
+        # Step 2: Check if credentials are available.
+        if not self._credentials or len(self._credentials) == 0:
+            return "No credentials found."
+
+        # Step 3: Calculate the maximum size for padding.
+        # This ensures proper alignment in the output.
+        max_size = max(
+            len(c.get("domain") or "") + len(c.get("username") or "")

```

Comment:
Can use `get()` default parmaeter
```python
len(c.get("domain", "")) + len(c.get("username", ""))
```

Reply:
In fact, no ! I tried but it failed too. I dig a little inside the problem and if the get return "", it could be interpreted as a None. It's better to put a or in order to ensure that it will be handled correctly in all cases.